src/Shared/SqlProcessor.cs

```diff
@@ -410,31 +455,39 @@ private static void ParseNextToken(
             var length = i - start;
             if (length > 0)
             {
-                var tokenSpan = sql.Slice(start, length);
-
                 // Special handling: if we are in a FROM clause, check if this identifier is a reserved keyword
                 // that indicates the end of the FROM clause.
                 if (state.InFromClause)
                 {
-                    foreach (var reservedKeyword in FromClauseReservedKeywords)
+                    var isReservedKeyword = false;
+
+                    // Range of shortest ("BY") to longest ("TABLESAMPLE") keywords
+                    if (length >= 2 && length <= 11)
```

Comment:
Would the compiler optimise this down to int constants so it could be slightly more self-explanatory?

```csharp
if (length >= "BY".Length && length <= "TABLESAMPLE".Length)
```

Reply:
I've taken a slightly different approach in my latest commit. We now compute this statically once based on the array values. That way, if we ever add anything, we don't need to remember to update the values.